library/core/src/cmp.rs

```diff
@@ -1410,11 +1411,18 @@ mod impls {
             impl const Ord for $t {
                 #[inline]
                 fn cmp(&self, other: &$t) -> Ordering {
-                    // The order here is important to generate more optimal assembly.
-                    // See <https://github.com/rust-lang/rust/issues/63758> for more info.
-                    if *self < *other { Less }
-                    else if *self == *other { Equal }
-                    else { Greater }
+                    #[cfg(bootstrap)]
+                    {
+                        // The order here is important to generate more optimal assembly.
+                        // See <https://github.com/rust-lang/rust/issues/63758> for more info.
+                        if *self < *other { Less }
+                        else if *self == *other { Equal }
+                        else { Greater }
+                    }
+                    #[cfg(not(bootstrap))]
+                    {
+                        crate::intrinsics::three_way_compare(*self, *other)
```

Comment:
a question for my own understanding: is there a reason why it's better (or has to be) for this to be in a single instance of the `fn cmp` rather than either `#[cfg(bootstrap)] fn cmp(...)` or on the impl?

Reply:
I just put it here because it's the least copy-pasting.  Doing it at other levels would work too, but would mean duplicating the `#[inline]` or stability attributes, which leaves more opportunities for me to get it wrong.

Reply:
You don't have the duplicate the signature and the `#[inline]` attribute. But it doesn't make that much of a difference (also hopefully we'll get rid of these `cfg(bootstrap)` inside the standard library soon anyways🤞)